nixos/modules/installer/tools/tools.nix

```diff
@@ -330,6 +331,15 @@ in
   config = {
     documentation.man.man-db.skipPackages = [ nixos-version ];
 
+    warnings = lib.optional (!config.system.disableInstallerTools && !config.system.rebuild.enableNg) ''
+      The Bash implementation of nixos-rebuild will be deprecated
+      and removed in the 26.05 release of NixOS. Please migrate to the
+      newer implementation by removing `system.rebuild.enableNg = false`
+      from your configuration. If you are unable to migrate due to any
+      issues with the new implementation, please create an issue and tag
+      the maintainers of `nixos-rebuild-ng`.
```

Comment:
I'm not sure if there's guidance written down on this either way,[^1] but note that warnings don't benefit from markdown rendering.

I believe we usually avoid line-wrapping in warning/assertion text, or rather make the line-wrapping explicitly reflect how you want the warning to be printed to stderr.

Not blocking the PR, just a small titbit of info.

[^1]: If there is guidance, I'd be interested in seeing it, if there isn't then where should it be added?